### ISSUE TITLE
V6 Bulidpack complains about read only filesystem

### ISSUE THREAD

**Issue by andycroll** [DataDog/heroku-buildpack-datadog#20]
Similar to #17 but I see a) a lot of logging on `heroku console` by default... and

b) This line. I figure the buildpack might be aware of Heroku's readonly-ness.

```
2017-11-22 17:36:20 UTC | ERROR | (apm.go:95 in func2) | seelog internal error: mkdir /var/log/datadog/: read-only file system
```

**Comment by jyee** [comment]
Thanks. I've raised this issue with our Agent team. It appears that the log file location isn't being passed to APM collector and needs to be updated in the apm.go file.

**Comment by webandtech** [comment]
Seeing this as well. It seems to be logging at INFO level and even when running a console.

```
2017-11-27 23:01:54 UTC | INFO | (start.go:176 in StartAgent) | GUI server port -1 specified: not starting the GUI.
2017-11-27 23:01:54 UTC | INFO | (forwarder.go:228 in Start) | DefaultForwarder started (4 workers), sending to 1 endpoint(s): "https://6-0-0-app.agent.datadoghq.com" (1 api key(s))
2017-11-27 23:01:54 UTC | ERROR | (tagger.go:67 in Init) | error initialising collector kubelet: Failed to connect to kubelet, Kubernetes tagging will not work: temporary failure in kubeutil, will retry later: Unable to get hostname from docker, please set the kubernetes_kubelet_host option: docker not available
2017-11-27 23:01:54 UTC | ERROR | (tagger.go:67 in Init) | error initialising collector docker: Failed to connect to Docker, docker tagging will not work: docker not available
2017-11-27 23:01:54 UTC | INFO | (udp.go:63 in Listen) | dogstatsd-udp: starting to listen on 127.0.0.1:8125
2017-11-27 23:01:54 UTC | ERROR | (tagger.go:67 in Init) | error initialising collector ecs: Failed to connect to ECS, ECS tagging will not work
2017-11-27 23:01:54 UTC | INFO | (runner.go:93 in NewRunner) | Runner started with 4 workers.
2017-11-27 23:01:54 UTC | INFO | (collector.go:51 in NewCollector) | Embedding Python 2.7.14 (default, Nov 22 2017, 14:10:55) [GCC 4.7.2]
```

**Comment by alexmic** [comment]
Same issue here.

**Comment by Fryie** [comment]
Is there a fix for this?

**Comment by justinkchen** [comment]
Still getting this error too

**Comment by alloy** [comment]
@jyee This error is adding a lot of noise to our logs when grepping for ‘error’. Could you please either prioritise fixing this or, if that’s not likely to happen soon and `apm.go` is open-source, point me to the relevant source so we can possibly provide a fix?

**Comment by webandtech** [comment]
Adding DD_LOG_TO_CONSOLE=false in our app environments solved this for us

**Comment by undergroundwebdesigns** [comment]
For anyone else seeing this, setting `DD_LOG_TO_CONSOLE=false` had no effect for us, but setting `DD_LOG_LEVEL` to `CRITICAL` seems to have worked.

**Comment by coffenbacher** [comment]
Perfect, thanks @undergroundwebdesigns . Works great for us.

**Comment by kevinohara80** [comment]
This is still an issue. Also, setting `DD_LOG_LEVEL` to `CRITICAL` has not effect. Neither does `DD_LOG_TO_CONSOLE` to `false`.

Here's the configuration:

```
DD_API_KEY:                xxxxxxxxxxxxxxxxxxxxxxxxxxxx
DD_DYNO_HOST:              true
DD_ENV:                    dev
DD_LOG_LEVEL:              CRITICAL
DD_LOG_TO_CONSOLE:         false
DD_SERVICE_ENV:            dev
DD_SERVICE_NAME:           my-heroku-app
DD_TAGS:                   env:dev platform:heroku
DD_TRACE_ENABLED:          true
```

Here are the logs received: 

```
2019-02-28T16:02:29.981032+00:00 app[web.1]: Starting Datadog Agent on my-heroku-app.web.1
2019-02-28T16:02:30.006735+00:00 app[web.1]: Starting Datadog Trace Agent on my-heroku-app.web.1
2019-02-28T16:02:30.203240+00:00 app[web.1]: 2019-02-28 16:02:30 INFO (config.go:214) -Loaded configuration: /app/.apt/etc/datadog-agent/datadog.yaml
2019-02-28T16:02:30.203703+00:00 app[web.1]: seelog internal error: mkdir /var/log/datadog/: read-only file system
```

**Comment by trr-aaronchu** [comment]
I had this issue when trying to use this feature: `DD_PROCESS_AGENT=true`. Once removed, the error with that path `/var/log/datadog` went away. However, in the documentation, it says the correct location for logs should be `/app/.apt/var/log/datadog` - so it seems like the Process Agent feature might not be picking up the right log location? If process tracking is available and supported, it would be great to use it.

**Comment by lancedikson** [comment]
Same here, just started trying trial and got the same error. Any updates on the fix, @jyee?

**Comment by agirlnamedsophia** [comment]
same here!

**Comment by arapulido** [comment]
This is now fixed in release `1.16`: https://github.com/DataDog/heroku-buildpack-datadog/releases/tag/1.16 if you use the agent pinned version (`6.18`, `7.18`) or newer
